src/services/projects.js

```diff
@@ -253,57 +264,65 @@ module.exports = class ProjectsHelper {
 	 * @returns {JSON} - project delete response.
 	 */
 
-	static async delete(resourceId, loggedInUserId) {
+	static async delete(resourceId, loggedInUserId, orgId) {
 		try {
-			const fetchOrgId = await resourceCreatorMappingQueries.findOne(
+			const resourceCreatorMapping = await resourceCreatorMappingQueries.findOne(
 				{
 					resource_id: resourceId,
 					creator_id: loggedInUserId,
 				},
 				['id', 'organization_id']
 			)
 
-			let fetchResourceId = null
-
-			if (fetchOrgId) {
-				fetchResourceId = await resourceQueries.findOne(
-					{
-						id: resourceId,
-						organization_id: fetchOrgId.organization_id,
-						status: common.STATUS_DRAFT,
-					},
-					{ attributes: ['id'] }
-				)
+			if (!resourceCreatorMapping?.id) {
+				throw new Error('PROJECT_NOT_FOUND')
 			}
 
-			if (!fetchOrgId || !fetchResourceId) {
-				return responses.failureResponse({
-					message: 'PROJECT_NOT_FOUND',
-					statusCode: httpStatusCode.bad_request,
-					responseCode: 'CLIENT_ERROR',
-				})
+			const resource = await resourceQueries.findOne(
+				{
+					id: resourceId,
+					organization_id: resourceCreatorMapping.organization_id,
+					status: common.RESOURCE_STATUS_DRAFT,
+				},
+				{ attributes: ['id', 'type', 'organization_id'] }
+			)
+
+			if (!resource?.id) {
+				throw new Error('PROJECT_NOT_FOUND')
 			}
 
-			let updatedProject = await resourceQueries.deleteOne(resourceId, fetchOrgId.organization_id)
+			let updatedProject = await resourceQueries.deleteOne(resourceId, resource.organization_id)
 			let updatedProjectCreatorMapping = await resourceCreatorMappingQueries.deleteOne(
-				fetchOrgId.id,
+				resourceCreatorMapping.id,
 				loggedInUserId
 			)
 
 			if (updatedProject === 0 && updatedProjectCreatorMapping === 0) {
-				return responses.failureResponse({
-					message: 'PROJECT_NOT_FOUND',
-					statusCode: httpStatusCode.bad_request,
-					responseCode: 'CLIENT_ERROR',
-				})
+				throw new Error('PROJECT_NOT_FOUND')
 			}
+
+			//add user action
+			eventBroadcaster(common.EVENT_ADD_USER_ACTION, {
```

Comment:
@priyanka-TL Rather then adding like this in multiple places, can we write a hook in model? E.g. - https://github.com/ELEVATE-Project/mentoring/blob/master/src/database/models/modules.js

Reply:
No we cant add this, hence update API is an auto save call, hence activities will add in each call.

Reply:
@priyanka-TL we can do for comments model and for resource model we can do the same by checking status, if status not equal to draft or something